packages/core/src/network/network-peers.ts

```diff
@@ -45,20 +40,31 @@ class NetworkPeers {
 
   // @returns a float between 0 (completely unreliable) and 1 (completely
   // reliable) estimating the quality of service of a peer's network connection
-  public qualityOf(peer: PeerId): number {
-    let entryIndex = this.findIndex(peer)
-    if (entryIndex >= 0 && this.peers[entryIndex].heartbeatsSent > 0) {
+  public qualityOf(peerId: PeerId): number {
+    const entry = this.peers.get(peerId.toB58String())
+    if (entry && entry.heartbeatsSent > 0) {
       /*
       return entry.heartbeatsSuccess / entry.heartbeatsSent
       */
-      return this.peers[entryIndex].lastTen
+      return entry.lastTen
     }
     return UNKNOWN_Q
```

Comment:
FYI: this will most likely change soon because of problems with fluent inavailability